File: reference/two-pointer/557-Reverse-Words-in-a-String-III.go
func reverseWords(s string) string {
\t// Convert string to a slice of characters (runes)
\tsList := []rune(s)
\t
\t// Function to reverse characters between indices l and h
\tutilReverseWord := func(l, h int) {
\t\tfor l < h {
\t\t\tsList[l], sList[h] = sList[h], sList[l]
\t\t\tl++
\t\t\th--
\t\t}
\t}
\t
\tlow := 0
\tfor high := 1; high < len(sList); high++ {
\t\tif sList[high] == ' ' {
\t\t\tutilReverseWord(low, high-1)
\t\t\tlow = high + 1
\t\t} else if high == len(sList)-1 {
\t\t\tutilReverseWord(low, len(sList)-1)
\t\t}
\t}
\t
\t// Convert the rune slice back to a string and return it
\treturn string(sList)
}

File: reference/two-pointer/557-Reverse-Words-in-a-String-III.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s_list = list(s)
        def util_reverword(l, h):
            while l < h:
                s_list[l], s_list[h] = s_list[h], s_list[l]
                l += 1
                h -= 1
            
        
        low = 0
        for high in range(1, len(s)):
            if s[high] == \ \:
                util_reverword(low, high - 1)
                low = high + 1
            elif high == len(s) - 1:
                util_reverword(low, len(s_list) - 1)
                
        return \\.join(s_list)
        


